### PR TITLE
Run hooks with websockets and rewrite ws headers

### PR DESCRIPTION
**Pull request by mcollina**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -159,21 +159,26 @@ An array that contains the types of the methods. Default: `['DELETE', 'GET', 'HE
 ## websocket
 
 This module has _partial_ support for forwarding websockets by passing a
-`websocket` option. All those options are going to be forwarded to
-[`@fastify/websocket`](https://github.com/fastify/fastify-websocket).
+`websocket` option.
```
Comment (Fdawgs) [inline README.md:162]: Can you clarify what the `websocket` option is? An object? An array? It is stated for the other options in the documentation.

File: README.md [hunk 2]
```diff
@@ -159,21 +159,26 @@ An array that contains the types of the methods. Default: `['DELETE', 'GET', 'HE
...
+2. support `ignoreTrailingSlash`
+3. forwarding more than one subprotocols. Note: Only the first subprotocol is being forwarded
 
 Pull requests are welcome to finish this feature.
 
+## `wsClientOptions`
+
+When specified `wsClientOptions` , the object is passed throught to the [`WebSocket` constructor](https://github.com/websockets/ws/blob/HEAD/doc/ws.md#class-websocket).
+
+It also support an additional `rewriteRequestHeaders(headers, request)` function that can be used to write the headers before
```
Comment (Fdawgs) [inline README.md:179]:
```suggestion
When specified, the `wsClientOptions` object is passed to the [`WebSocket` constructor](https://github.com/websockets/ws/blob/HEAD/doc/ws.md#class-websocket).

It also supports an additional `rewriteRequestHeaders(headers, request)` function that can be used to write the headers before
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]
Few tweaks/questions

File: README.md (hunk 1 above)
Comment (mcollina) [reply README.md:162]: I've improved the docs.

**Review by Fdawgs — APPROVED** [review]
LGTM on the docs front. Don't know enough about websockets to comment on functionality, but will leave that to the others to review!

**Comment by mcollina** [comment]
Note that this also fixes a regression for the getUpstream() support for Websocket that was changed in https://github.com/fastify/fastify-reply-from/pull/293

**Review by Eomm — COMMENTED** [review]
minor things - logic 

Do you agree to release it as major?

File: index.js [hunk 3]
```diff
@@ -117,66 +159,71 @@ class WebSocketProxy {
...
     const { target: url, wsClientOptions } = upstream
+    let rewriteRequestHeaders = defaultWsHeadersRewrite
+    let headersToRewrite = {}
+
+    if (wsClientOptions && wsClientOptions.headers) {
+      headersToRewrite = wsClientOptions.headers
+    }
+    if (wsClientOptions && wsClientOptions.rewriteRequestHeaders) {
+      rewriteRequestHeaders = wsClientOptions.rewriteRequestHeaders
+    }
```
[the remark this answers is not in the document]
Comment (mcollina) [reply index.js:177]: I would need to drop support for Node v14. I think it's ok, right?

**Comment by mcollina** [comment]
Yes, this should be a major.

File: index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mcollina) [reply index.js]: Actually no, this worked in v14

**Review by Eomm — APPROVED** [review]

**Pull request merged**